osu.Game/Skinning/LegacyManiaSkinDecoder.cs

```diff
@@ -128,34 +128,36 @@ private void flushPendingLines()
                         currentConfig.LightFramePerSecond = lightFramePerSecond > 0 ? lightFramePerSecond : 24;
                         break;
 
-                    case string when pair.Key.StartsWith("Colour", StringComparison.Ordinal):
+                    case { } when pair.Key.StartsWith("Colour", StringComparison.Ordinal):
                         HandleColours(currentConfig, line, true);
                         break;
 
                     // Custom sprite paths
-                    case string when pair.Key.StartsWith("NoteImage", StringComparison.Ordinal):
-                    case string when pair.Key.StartsWith("KeyImage", StringComparison.Ordinal):
-                    case string when pair.Key.StartsWith("Hit", StringComparison.Ordinal):
-                    case string when pair.Key.StartsWith("Stage", StringComparison.Ordinal):
-                    case string when pair.Key.StartsWith("Lighting", StringComparison.Ordinal):
-                        currentConfig.ImageLookups[pair.Key] = pair.Value;
+                    case { } when pair.Key.StartsWith("NoteImage", StringComparison.Ordinal):
+                    case { } when pair.Key.StartsWith("KeyImage", StringComparison.Ordinal):
+                    case { } when pair.Key.StartsWith("Hit", StringComparison.Ordinal):
+                    case { } when pair.Key.StartsWith("Stage", StringComparison.Ordinal):
+                    case { } when pair.Key.StartsWith("Lighting", StringComparison.Ordinal):
+                        currentConfig.ImageLookups[pair.Key.ToString()] = pair.Value.ToString();
                         break;
                 }
             }
 
             pendingLines.Clear();
         }
 
-        private void parseArrayValue(string value, float[] output, bool applyScaleFactor = true)
+        private void parseArrayValue(ReadOnlySpan<char> value, float[] output, bool applyScaleFactor = true)
         {
-            string[] values = value.Split(',');
+            int valuesCount = value.Count(',') + 1;
+            Span<Range> ranges = valuesCount <= 256 ? stackalloc Range[valuesCount] : new Range[valuesCount];
```

Comment:
https://github.com/dotnet/runtime/issues/934 has just been approved for .NET 9. Should we adapt that API for splitting into undetermined counts?

Reply:
Are you suggesting to pull the in-progress `SpanSplitEnumerator`? I think we can go with this as it is for now and use the .NET 9 API when it's formally released. We will probably be upgrading to .NET 9 early anyway.

Reply:
Yes, I meant to add a naive implementation with the same API shape so that adapting will be easy. It should be fine to do it later.